x-pack/legacy/plugins/monitoring/server/alerts/license_expiration.ts

```diff
@@ -0,0 +1,161 @@
+  ccsEnabled: boolean
+): AlertType => {
+  async function getCallCluster(services: any): Promise<any> {
+    const monitoringCluster = await getMonitoringCluster();
+    if (!monitoringCluster) {
+      return services.callCluster;
+    }
+
+    return monitoringCluster.callCluster;
+  }
+
+  const logger = getLogger([ALERT_TYPE_LICENSE_EXPIRATION]);
+  return {
+    id: ALERT_TYPE_LICENSE_EXPIRATION,
+    name: 'Monitoring Alert - License Expiration',
+    actionGroups: ['default'],
+    async executor({
+      services,
+      params,
+      state,
+    }: LicenseExpirationAlertExecutorOptions): Promise<any> {
+      logger.debug(
+        `Firing alert with params: ${JSON.stringify(params)} and state: ${JSON.stringify(state)}`
+      );
+      const { dateFormat, timezone } = params as AlertParams;
+
+      const callCluster = await getCallCluster(services);
+
+      // Support CCS use cases by querying to find available remote clusters
+      // and then adding those to the index pattern we are searching against
+      let esIndexPattern = INDEX_PATTERN_ELASTICSEARCH;
+      if (ccsEnabled) {
+        const availableCcs = await fetchAvailableCcs(callCluster);
+        if (availableCcs.length > 0) {
+          esIndexPattern = getCcsIndexPattern(esIndexPattern, availableCcs);
+        }
+      }
+
+      const clusters = await fetchClusters(callCluster, esIndexPattern);
+
+      // Fetch licensing information from cluster_stats documents
+      const licenses: AlertLicense[] = await fetchLicenses(callCluster, clusters, esIndexPattern);
+      if (licenses.length === 0) {
+        logger.warn(`No license found for ${ALERT_TYPE_LICENSE_EXPIRATION}.`);
+        return state;
+      }
+
+      const uiSettings = server.newPlatform.__internals.uiSettings.asScopedToClient(
+        services.savedObjectsClient
+      );
+      const emailAddress = await fetchDefaultEmailAddress(uiSettings);
+      if (!emailAddress) {
+        // TODO: we can do more here
+        logger.warn(
+          `Unable to send email for ${ALERT_TYPE_LICENSE_EXPIRATION} because there is no email configured.` +
+            ` Please configure 'xpack.monitoring.cluster_alerts.email_notifications.email_address'.`
+        );
+        return;
+      }
+
+      const result: AlertState = { ...state };
+
+      for (const license of licenses) {
+        const licenseState: AlertClusterState = state[license.clusterUuid] || {};
+        const $expiry = moment.utc(license.expiryDateMS);
+        let isExpired = false;
+        let severity = 0;
+
+        if (license.status !== 'active') {
+          isExpired = true;
+          severity = 2001;
+        } else if (license.expiryDateMS) {
+          for (let i = EXPIRES_DAYS.length - 1; i >= 0; i--) {
+            if (license.type === 'trial' && i < 2) {
+              break;
+            }
+
+            const $fromNow = moment().add(EXPIRES_DAYS[i], 'days');
+            if ($fromNow.isAfter($expiry)) {
```

Comment:
Did you mean to compare utc time with the local time here? Maybe cast both as utc (just in case)?:
```javascript
const $fromNow = moment.utc(Date.now()).add(EXPIRES_DAYS[i], 'days');
```

Maybe does not matter much, since most servers are utc, but just to be sure

Reply:
Good point, I'll do `moment.utc().add(...` to be sure. Thanks!